crates/sparrow-expressions/src/evaluators/json_field.rs

```diff
@@ -0,0 +1,154 @@
+use std::sync::Arc;
+
+use arrow_array::StringArray;
+
+use crate::evaluator::Evaluator;
+use crate::values::StringValue;
+
+inventory::submit!(crate::evaluators::EvaluatorFactory {
+    name: "json_field",
+    create: &create
+});
+
+/// Evaluator for `json_field` expressions.
+///
+/// This evaluator expects a `string` input and `field` name.
+/// It parses the `string` into a `json` object, then outputs the
+/// value of the `field` name as a `string`, or `null` if the `field`
+/// does not exist.
```

Comment:
From what I can tell this code deals with a single JSON object and a path of length 1. So the for example following is *not* supported 

```
// JSON, path
{ a: { b : 1}, c: 10} , a.b
```

But I think this works? 

```
// JSON, path
{ a: { b : 1}, c: 10} , a
```

Reply:
Yeah, you can do `json_field(json_field(..., "a"), "b")` to do nesting. Ideally we'd do more with paths and/or parsing these into real JSON objects for manipulation instead of going back and forth with strings, but this was initial / rudimentary support.

Reply:
I think @jordanrfrazier may be able to comment and/or we should import some more notes on this -- there was a plan for how this could evolve to reduce the parsing. I think it was something like if we had a JSON object representation, then this could be `json_field(s, "field") = parse_json(s).field`, but I could be mistaken.